src/Expensify.js

```diff
@@ -102,28 +100,41 @@ class Expensify extends PureComponent {
     componentDidUpdate(prevProps) {
         const previousAccountID = lodashGet(prevProps, 'session.accountID', null);
         const currentAccountID = lodashGet(this.props, 'session.accountID', null);
+
         if (currentAccountID && (currentAccountID !== previousAccountID)) {
             PushNotification.register(currentAccountID);
         }
 
-        // If we previously had no authToken and now have an authToken we'll want to reshow the boot splash screen so
-        // that we can remove it again once the content is ready
-        const previousAuthToken = lodashGet(prevProps, 'session.authToken', null);
-        if (this.getAuthToken() && !previousAuthToken) {
-            this.showSplash();
-        }
+        if (this.state.isNavigationReady && this.state.isSplashShown) {
+            const authStackReady = this.props.initialReportDataLoaded && this.props.isSidebarLoaded;
+            const shouldHideSplash = !this.isAuthenticated() || authStackReady;
 
-        if (this.getAuthToken() && this.props.initialReportDataLoaded && this.props.isSidebarLoaded) {
-            this.hideSplash();
+            if (shouldHideSplash) {
+                this.hideSplash();
+            }
         }
     }
 
     componentWillUnmount() {
         AppState.removeEventListener('change', this.initializeClient);
     }
 
-    getAuthToken() {
-        return lodashGet(this.props, 'session.authToken', null);
+    setNavigationReady() {
+        this.setState((state) => {
+            if (state.isNavigationReady) {
+                return state;
+            }
+
+            return {
+                ...state,
+                isNavigationReady: true,
+            };
+        });
+    }
```

Comment:
This is the same as

```js
setNavigationReady() {
  this.setState({ isNavigationReady: true });
}
```

When the state is already `true` it would just do nothing

But do we expect multiple `onReady` calls to be triggered anyway?

Reply:
Didn't see that it was a `PureComponent`, so it performs shallow compare on states comparaison anyway indeed. And lie you say, it shouldn't be called more than once.